src/gen/functions.rs

```diff
@@ -1,59 +1,150 @@
+
 use genco::prelude::*;
-use uniffi_bindgen::interface::{AsType, Function};
+use uniffi_bindgen::backend::Type;
+use uniffi_bindgen::interface::{AsType, Callable, ExternalKind, Function};
 
 use crate::gen::oracle::DartCodeOracle;
 use crate::gen::render::AsRenderable;
 
+use super::oracle::AsCodeType;
 use super::render::TypeHelperRenderer;
-use super::types::{generate_ffi_dart_type, generate_ffi_type};
-use super::utils::{fn_name, var_name};
-
-#[allow(unused_variables)]
-pub fn generate_function(
-    api: &str,
-    fun: &Function,
-    type_helper: &dyn TypeHelperRenderer,
-) -> dart::Tokens {
-    let ffi = fun.ffi_func();
-    let fn_name = fn_name(fun.name());
-    let args = quote!($(for arg in &fun.arguments() => $(&arg.as_renderable().render_type(&arg.as_type(), type_helper)) $(var_name(arg.name())),));
-    let ff_name = ffi.name();
-    let inner = quote! {
-    rustCall(api, (res) =>
-        _$(&fn_name)(
-            $(for arg in &fun.arguments() => $(DartCodeOracle::type_lower_fn(&arg.as_type(), quote!($(var_name(arg.name()))))),)
-        res)
-    )
-    };
 
-    let (ret, body) = if let Some(ret) = fun.return_type() {
+// #[allow(unused_variables)]
```

Comment:
What speaks against just removing this? We can still recover it from the git history if we really need to (but in my experience this is very unlikely to happen ;) ).